spock-core/src/main/java/org/spockframework/runtime/model/FeatureInfo.java

```diff
@@ -171,6 +174,14 @@ public List<FeatureInfo> getImpliedFeatures() {
    */
   @Beta
   public void addImpliedFeature(FeatureInfo feature) {
+    if (equals(feature)) {
+      throw new IllegalArgumentException("Features cannot imply themselves");
+    }
+
+    if (!getParent().equals(feature.getParent())) {
+      throw new IllegalArgumentException("Features can only imply features within the same specification");
```

Comment:
I think you need to be a bit more flexible here, I'm pretty sure that features will have the Spec they are declared on as parent, with this it would only accept other features on the same hierarchy level. This is fine for `@Stepwise` as it doesn't support inheritance, but it might be overly limiting.
Could you add a test that checks that `impliedFeature` works from `Parent` -> `Child` and vice-versa.
This should only be an issue on the `*Info` and not the `*Node` level.

Reply:
Yep, good point.
Changed.

Reply:
Yep, good point.
Changed.